Sources/Operator/py_operator.hpp

```diff
@@ -172,6 +173,56 @@ void AddOperatorModule(py::module m) {
 
             Returns:
                 A numpy array of local values of the operator.)EOF");
+
+  subm.def(
+      "der_local_values",
+      [](AbstractOperator& op, AbstractMachine& machine,
+         py::array_t<double, py::array::c_style> samples, Index batch_size) {
+        switch (samples.ndim()) {
+          case 3: {
+            auto local_values = py::cast(DerLocalValues(
+                Eigen::Map<const RowMatrix<double>>{
+                    samples.data(), samples.shape(0) * samples.shape(1),
+                    samples.shape(2)},
+                machine, op, batch_size));
+            local_values.attr("resize")(samples.shape(0), samples.shape(1),
+                                        samples.shape(2));
+            return local_values;
+          }
+          case 2:
+            return py::cast(DerLocalValues(
+                Eigen::Map<const RowMatrix<double>>{
+                    samples.data(), samples.shape(0), samples.shape(1)},
```

Comment:
shouldn't there be the number of parameters in the shape somewhere?

Reply:
yes, thanks